lisa/tools/grub_config.py

```diff
@@ -0,0 +1,172 @@
+from lisa.util import UnsupportedDistroException
+
+if TYPE_CHECKING:
+    from lisa.node import Node
+    from lisa.operating_system import Posix
+
+
+class GrubConfig(Tool):
+    @classmethod
+    def create(cls, node: "Node", *args: Any, **kwargs: Any) -> Tool:
+        """
+        Factory method to create an instance of the Grub tool.
+        """
+        if isinstance(node.os, CBLMariner):
+            if node.os.information.release == "2.0":
+                return GrubConfigAzl2(node, args, kwargs)
+            if node.os.information.release == "3.0":
+                return GrubConfigAzl3(node, args, kwargs)
+
+        raise UnsupportedDistroException(
+            os=node.os, message="Grub tool only supported on CBLMariner 2.0 and 3.0."
+        )
+
+    def __init__(
+        self, command: str, package: str, node: "Node", *args: Any, **kwargs: Any
+    ) -> None:
+        super().__init__(node, *args, **kwargs)
+        self._command = command
+        self._package = package
+
+    @property
+    def command(self) -> str:
+        return self._command
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def set_fips_mode(self, fips_mode: bool) -> None:
+        """
+        Set the FIPS mode to the specified value.
+        """
+        raise NotImplementedError("set_fips_mode is not implemented.")
+
+    def set_boot_uuid(self, uuid: str, same_as_root: bool) -> None:
+        """
+        Set the boot UUID to the specified value.
+        """
+        if same_as_root:
+            self.remove_kernel_cmdline_arg(r"boot")
+        else:
+            self.set_kernel_cmdline_arg(f"boot=UUID={uuid}")
+
+    def unset_boot_uuid(self, same_as_root: bool) -> None:
+        """
+        Unset the boot UUID.
+        """
+        raise NotImplementedError("unset_boot_uuid is not implemented.")
+
+    def remove_kernel_cmdline_arg(self, arg: str) -> None:
+        """
+        Remove the specified kernel command line argument from the grub configuration.
+        """
+        raise NotImplementedError("remove_kernel_cmdline_arg is not implemented.")
+
+    def set_kernel_cmdline_arg(self, arg: str) -> None:
+        """
+        Append the specified kernel command line argument to the grub configuration.
+        """
+        raise NotImplementedError("set_kernel_cmdline_arg is not implemented.")
+
+    def apply(self) -> None:
+        """
+        Reconfigure grub to apply the changes made to the kernel command line arguments.
+        """
+        raise NotImplementedError("apply is not implemented.")
+
+    def _install(self) -> bool:
+        posix_os: Posix = self.node.os  # type: ignore
+        posix_os.install_packages(self._package)
+        return self._check_exists()
+
+
+class GrubConfigAzl2(GrubConfig):
+    def __init__(self, node: "Node", *args: Any, **kwargs: Any) -> None:
+        super().__init__("grubby", "grubby", node, *args, **kwargs)
+
+    def set_fips_mode(self, fips_mode: bool) -> None:
```

Comment:
The GrubConfig tool should provide functionality itself, and not provide high level functionality. So the `set_fips_mode` should be moved to fips level.